bin/nib-ls

```diff
@@ -214,18 +227,43 @@ def proc_file(f, opts):
         else:
             row += ['error']
 
-    if opts.stats:
+    if opts.stats or opts.counts:
         # We are doomed to load data
         try:
             d = vol.get_data()
             if not opts.stats_zeros:
                 d = d[np.nonzero(d)]
-            # just # of elements
-            row += ["[%d] " % np.prod(d.shape)]
-            # stats
-            row += [len(d) and '%.2g:%.2g' % (np.min(d), np.max(d)) or '-']
-        except Exception as e:
-            verbose(2, "Failed to obtain stats -- %s" % str(e))
+            else:
+                # at least flatten it -- functionality below doesn't
+                # depend on the original shape, so let's use a flat view
+                d = d.reshape(-1)
+            if opts.stats:
+                # just # of elements
+                row += ["@l[%d]" % np.prod(d.shape)]
+                # stats
+                row += [len(d) and '@l[%.2g, %.2g]' % (np.min(d), np.max(d)) or '-']
+            if opts.counts:
+                maxunique = 1000  # constant
+                # Since d might be quite large and we don't want to exhaust memory and
+                # thrush, let's do in manageable batches
```

Comment:
Or, I guess - go back to the counter implementation and barf out when you get to too many unique values?